### PR TITLE
FORMS-530: [Regression] CSV export only exports one iteration value

### PR DESCRIPTION
**Pull request by eahuma**

### CODE REVIEW

File: src/java/fr/paris/lutece/plugins/forms/export/csv/CSVDataLine.java [hunk 1]
```diff
@@ -100,8 +100,16 @@ public void addData( FormQuestionResponse formQuestionResponse )
...
-
-        _mapDataToExport.put( Pair.of( question.getId( ), question.getIterationNumber( ) ), sbReponseValues.toString( ) );
+        if ( !_mapDataToExport.containsKey( question.getId( ) ) )
+        {
+            _mapDataToExport.put( question.getId( ), sbReponseValues.toString( ) );
+        }
+        else
+        {
+            StringBuilder sbConcatReponseValues = new StringBuilder ( );
+            sbConcatReponseValues.append( _mapDataToExport.get( question.getId( ) ) ).append( sbReponseValues.toString( ) );
```
Comment (rzara) [inline src/java/fr/paris/lutece/plugins/forms/export/csv/CSVDataLine.java:110]: There should be a separator between values

**Review by rzara — COMMENTED** [review]
This change seems to collect all values in a single column. This is a regression from the previous forms versions, and may make the resulting file difficult to exploit. For instance, what if in a group, questions are optionnal, and there are at least 3 iterations. You would not be able to regroup each iteration values together.

File: src/java/fr/paris/lutece/plugins/forms/export/csv/CSVDataLine.java [hunk 2]
```diff
@@ -100,8 +100,16 @@ public void addData( FormQuestionResponse formQuestionResponse )
...
-
-        _mapDataToExport.put( Pair.of( question.getId( ), question.getIterationNumber( ) ), sbReponseValues.toString( ) );
+        if ( !_mapDataToExport.containsKey( question.getId( ) ) )
+        {
+            _mapDataToExport.put( question.getId( ), sbReponseValues.toString( ) );
+        }
+        else
+        {
+            StringBuilder sbConcatReponseValues = new StringBuilder ( );
+            sbConcatReponseValues.append( _mapDataToExport.get( question.getId( ) ) ).append( ITERATION_SEPARATOR ).append( sbReponseValues.toString( ) );
```
Comment (rzara) [inline src/java/fr/paris/lutece/plugins/forms/export/csv/CSVDataLine.java:110]: What if the response value contains `ITERATION_SEPARATOR` ?

**Review by rzara — COMMENTED** [review]
How does this handle the following case:
|    | it1    | it2    | it3    |   |
|----|--------|--------|--------|---|
| q1 | val1_1 | val1_2 | val1_3 |   |
| q2 |        | foo    |        |   |
| q3 |        |        | bar    |   |

how will you know to which value of `q1` `foo` and `bar` are to be attached ?

**Comment by ThomasDumont** [comment]
A squashed commit was added to branch develop.
New jira must be created : 
- to study the possibility to have one column per iteration (as in forms v1)
- to check whether empty responses are taken into account in the export
- to check whether responses are escaped before being exported in csv

**Pull request closed without merging**